pkg/server/apiextensions.go

```diff
@@ -91,12 +99,13 @@ func (c *inheritanceCRDLister) ListWithContext(ctx context.Context, selector lab
 
 		if workspace != nil && workspace.Spec.InheritFrom != "" {
 			// Make sure the source workspace exists
-			sourceWorkspaceKey := clusters.ToClusterAwareKey(adminClusterName, workspace.Spec.InheritFrom)
+			sourceWorkspaceKey := clusters.ToClusterAwareKey(org, workspace.Spec.InheritFrom)
 			_, err := c.workspaceLister.Get(sourceWorkspaceKey)
 			switch {
 			case err == nil:
 				inheriting = true
-				inheritFrom = workspace.Spec.InheritFrom
+				// FIXME use a helper
```

Comment:
Somehow this seems like it should not leak outside of the helper lib?

Reply:
Fixed